File: Week 02/id_712/Leetcode_144.py
class Solution:
    def preorderTraversal(self,root):
        if not root:
            return []
        preorder=[]
        stack=[root]
        while stack:
            node=stack.pop()
            preorder.append(node.val)
            if node.right:
                stack.append(node.right)
            if node.left:
                stack.append(node.left)
        return preorder

File: Week 02/id_292/leetcode144.java
package com.angle.datastructure.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Autor:hash
 * Date: 2019/10/27
 * Description:
 */
public class TreeNode {
    private int val;
    private TreeNode left;
    private TreeNode right;

    public TreeNode(int x) {
        val = x;
    }

    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> list = new ArrayList<>();
        getNode(root, list);
        return list;
    }

    private void getNode(TreeNode root, List<Integer> list) {
        if (root != null) {
            list.add(root.val);
            if (root.left != null) {
                getNode(root.left, list);
            }
            if (root.right != null) {
                getNode(root.right, list);
            }
        }
    }
}
